plugins/rocketmq/src/main/java/com/navercorp/pinpoint/plugin/rocketmq/interceptor/ProducerSendCallBackInterceptor.java

```diff
@@ -0,0 +1,106 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.navercorp.pinpoint.plugin.rocketmq.interceptor;
+
+import org.apache.rocketmq.client.producer.SendResult;
+import org.apache.rocketmq.client.producer.SendStatus;
+
+import com.navercorp.pinpoint.bootstrap.async.AsyncContextAccessor;
+import com.navercorp.pinpoint.bootstrap.context.AsyncContext;
+import com.navercorp.pinpoint.bootstrap.context.MethodDescriptor;
+import com.navercorp.pinpoint.bootstrap.context.SpanEventRecorder;
+import com.navercorp.pinpoint.bootstrap.context.TraceContext;
+import com.navercorp.pinpoint.bootstrap.interceptor.AsyncContextSpanEventSimpleAroundInterceptor;
+import com.navercorp.pinpoint.bootstrap.interceptor.SpanEventSimpleAroundInterceptorForPlugin;
+import com.navercorp.pinpoint.plugin.rocketmq.RocketMQConstants;
+
+/**
+ * @author messi-gao
+ */
+public final class ProducerSendCallBackInterceptor {
+    public static class ConstructInterceptor extends SpanEventSimpleAroundInterceptorForPlugin {
+        public ConstructInterceptor(TraceContext traceContext, MethodDescriptor descriptor) {
+            super(traceContext, descriptor);
+        }
+
+        @Override
+        protected void doInBeforeTrace(SpanEventRecorder recorder, Object target, Object[] args) {
+            final AsyncContext asyncContext = recorder.recordNextAsyncContext();
+            ((AsyncContextAccessor) target)._$PINPOINT$_setAsyncContext(asyncContext);
+        }
+
+        @Override
+        protected void doInAfterTrace(SpanEventRecorder recorder, Object target, Object[] args, Object result,
+                                      Throwable throwable) {
+            recorder.recordServiceType(RocketMQConstants.ROCKETMQ_CLIENT_INTERNAL);
+            recorder.recordApi(methodDescriptor);
+            recorder.recordException(throwable);
+        }
+    }
+
+    public static class OnSuccessInterceptor extends AsyncContextSpanEventSimpleAroundInterceptor {
+        public OnSuccessInterceptor(
+                TraceContext traceContext, MethodDescriptor methodDescriptor) {
+            super(traceContext, methodDescriptor);
+        }
+
+        @Override
+        protected void doInBeforeTrace(SpanEventRecorder spanEventRecorder, AsyncContext asyncContext, Object o,
+                                       Object[] objects) {
+        }
+
+        @Override
+        protected void doInAfterTrace(SpanEventRecorder spanEventRecorder, Object target, Object[] args,
+                                      Object result, Throwable throwable) {
+            spanEventRecorder.recordServiceType(RocketMQConstants.ROCKETMQ_CLIENT_INTERNAL);
+            spanEventRecorder.recordApi(methodDescriptor);
+            final SendStatus sendStatus = ((SendResult) args[0]).getSendStatus();
+            if (throwable != null) {
```

Comment:
Like HttpStatusCode, how about using the AnnotationKey to record the SendStatus?
If an exception occurs when executing CallBack code, it is better if the user knows that callback has what kind of SendStatus occurred at the time the exception occurred.

In addition, if there is no exception in the callback, it seems okay to execute a recordException that is not SEND_OK as it is now.

Reply:
> Like HttpStatusCode, how about using the AnnotationKey to record the SendStatus?
> If an exception occurs when executing CallBack code, it is better if the user knows that callback has what kind of SendStatus occurred at the time the exception occurred.
> 
> In addition, if there is no exception in the callback, it seems okay to execute a recordException that is not SEND_OK as it is now.

```
        recorder.recordApi(methodDescriptor);
        final ConsumeConcurrentlyStatus status = (ConsumeConcurrentlyStatus) result;
        recorder.recordAttribute(RocketMQConstants.ROCKETMQ_SEND_STATUS_ANNOTATION_KEY, status.name());
        if (throwable != null) {
            recorder.recordException(throwable);
        }
```
like this?

Reply:
Yes, It looks good. :) 

But, judging method with watching interface, result will be null. 
=>
`final ConsumeConcurrentlyStatus status = (ConsumeConcurrentlyStatus) result;`

Reply:
It looks good at having changed Consumer side. :)
Please check about Producer side in ProducerSendCallBackInterceptor.

Reply:
It looks good at having changed Consumer side. Please check about Producer side in ProducerSendCallBackInterceptor.

Reply:
> It looks good at having changed Consumer side. :)
> Please check about Producer side in ProducerSendCallBackInterceptor.

fix it.

Reply:
Please check it again. It seems that it did not be changed.

Reply:
Please check it again. It seems that it did not be changed.

Reply:
I'am sorry.I fixed it just now.